src/components/CreateShoppingList.jsx

```diff
@@ -0,0 +1,38 @@
+import { useState } from 'react';
+import { createList } from '../api/firebase';
+import { useNavigate } from 'react-router-dom';
+import toast from 'react-hot-toast';
+
+export default function CreateShoppingList({ user, setListPath }) {
+	const [listName, setListName] = useState('');
+	const userId = user?.uid;
+	const userEmail = user?.email;
+	const navigate = useNavigate();
+
+	const handleSubmit = async (e) => {
+		e.preventDefault();
+
+		try {
+			const listDocRef = await createList(userId, userEmail, listName);
+			setListPath(listDocRef);
```

Comment:
Just a small detail, but the variable name `listDocRef` might be a bit misleading since `createList` actually returns the list path. Renaming it to `listPath`or something similar could make things clearer.